### DISCUSSION TITLE
what cuda toolkit version are supported?

### DISCUSSION THREAD

**Discussion by pivolan** [coqui-ai/TTS#2441] [General Q&A] [answered]
Tried to install in docker container but with new cuda train does not start. 
What version of cuda toolkit should I install?
now I have cuda 12.0.

**Comment by pivolan** [comment]
<img width="1276" alt="image" src="https://user-images.githubusercontent.com/903550/226731608-abfe5430-95c4-4b97-8b09-c3d2e160ee4a.png">
Tried different ways to set up the whole environment. Used docker images with nvidia pytorch, also tried from this repository. The docker container has no drivers and the right libraries, after installing them with the command: 
apt install cuda
The workout runs, but hangs on the first epoch and nothing happens after that. The phoneme generation step is successful.
Tried several images for the docker, but nothing helps, the result is the same. 
Here is the output of the nvidia-smi command

**Comment by pivolan** [comment]
in google colab training starts well.

**Comment by pivolan** [comment]
nvidia-smi results:
<img width="622" alt="image" src="https://user-images.githubusercontent.com/903550/226733520-3c9ab20b-e177-4d17-99e0-ce7b01f5ad69.png">

**Reply by p0p4k** [reply]
Hi, my system has cuda12 and is successfully running tts.

**Comment by erogol** [comment]
We support whatever pytorch supports. But there is some legacy code. But don't worry about it until it raised an error :)

**Comment by pivolan** [comment, marked as answer]
was fixed by this thread: 
https://github.com/coqui-ai/TTS/issues/2449#issuecomment-1494897168
thanks
